view/src/components/foodbanks.js

```diff
@@ -0,0 +1,937 @@
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
+  },
+  search: {
+    position: "relative",
+    borderRadius: theme.shape.borderRadius,
+    marginLeft: 0,
+    width: "100%",
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: "100%",
+    position: "absolute",
+    pointerEvents: "none",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  inputRoot: {
+    color: "inherit",
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    width: "100%",
+  },
+  chip: {
+    margin: "4px",
+  },
+});
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+/**
+ * This function sets up the mask used for the phone input of (***) ***-****
+ * where * represents a digit
+ */
+function TextMaskCustom(props) {
+  const { inputRef, ...other } = props;
+
+  return (
+    <MaskedInput
+      {...other}
+      ref={(ref) => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={[
+        "(",
+        /[1-9]/,
+        /\d/,
+        /\d/,
+        ")",
+        " ",
+        /\d/,
+        /\d/,
+        /\d/,
+        "-",
+        /\d/,
+        /\d/,
+        /\d/,
+        /\d/,
+      ]}
+      placeholderChar={"\u2000"}
+      showMask
+    />
+  );
+}
+
+TextMaskCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+};
+
+/**
+ * This class represents a Food Bank component, which is a sub-page of the
+ * home page where food bank objects are visualized, created, updated, edited,
+ * and deleted.
+ */
+class Foodbank extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // food bank states
+      foodbanks: "",
+      foodbankName: "",
+      location: "",
+      locationId: "",
+      hours: "",
+      foodbankId: "",
+      contactPhone: "(1  )    -    ",
+      contactName: "",
+      contactEmail: "",
+      forklift: false,
+      pallet: false,
+      loadingDock: false,
+      maxLoadSize: "",
+      refrigerationSpaceAvailable: "",
+      foodbankTags: [],
+      // page states
+      errors: [],
+      open: false,
+      uiLoading: true,
+      buttonType: "",
+      viewOpen: false,
+    };
+
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleEditClick = this.handleEditClick.bind(this);
+    this.handleViewOpen = this.handleViewOpen.bind(this);
+  }
+
+  /**
+   * Given an event, this function updates a state (the target of the event)
+   * with a new value
+   * @param event The event that is attempting to update a state
+   */
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  /** Used to uncheck and check the form checkboxes */
+  handleChecked = (name) => (event) => {
+    this.setState({ ...this.state, [name]: event.target.checked });
```

Comment:
If setState automatically merges, why do you need `...this.state` here?

Reply:
So, I did a little investigation. It appears that react will carry out shallow merging of states, but we are required to include the current state when we are changing a state within another state object. Here is the nice medium article that I read: https://medium.com/@imrobinkim/how-state-updates-are-merged-in-react-e07fc669fec2. 

As a result, I think the ...this.state is completely unnecessary since the state is not nested in an object!

Reply:
Ah, awesome!  Good thing you didn't listen to me when I said you should introduce a nested state object for your dialog state then, huh?  :)